Add utility to write timestamped CSV performance reports

Srishti-8976 opened the pull request:
This PR adds a timestamped CSV performance report per run and integrates it into the existing perfTestRunner without affecting existing JSON outputs.

Srishti-8976 committed:
Add utility to write timestamped CSV performance reports

Srishti-8976 committed:
Integrate CSV writer into perfTestRunner

Srishti-8976 committed:
Integrate CSV writer into perfTestRunner

sonalgoyal commented on perfTestRunner.py:
```diff
@@ -154,6 +156,14 @@ def perform_load_test():
     # Save results after successful test execution
     save_results(test_data)
 
+    if "train" in test_data["results"] and "match" in test_data["results"]:
```
can you make this generic please? so if we have other tests, it just loops through them and creates the appropriate column?

sonalgoyal replied:
also, can we have reports per test per year? that may be easier to send into the visualization.

Srishti-8976 replied:
Thanks for the feedback! I’ve updated the implementation to make CSV generation fully generic by dynamically iterating over all phases, and organized reports under perf_reports/testName/year. The runner now passes the complete results dictionary to the CSV writer. Please let me know if this looks good.

sonalgoyal replied:
@padam-prakash please review

Srishti-8976 committed:
Make CSV performance reporting generic and organize reports by test/year

Srishti-8976 committed:
Update perfTestRunner to write generic CSV performance reports

Srishti-8976 committed:
Create run-performance.yml

Srishti-8976 committed:
Fix workflow to run perfTestRunner.py

Srishti-8976 committed:
Add results directory for CSV outputs

Srishti-8976 committed:
Enable automatic saving of performance CSV results to GitHub

Srishti-8976 committed:
Configure workflow to run FEBRL performance test and auto-save CSV results

Srishti-8976 committed:
Update run-performance.yml

Srishti-8976 committed:
Update run-performance.yml

Srishti-8976 committed:
Update run-performance.yml

Srishti-8976 committed:
Update perf_csv_writer.py

Srishti-8976 committed:
Update perfTestRunner.py

Srishti-8976 committed:
Update run-performance.yml

Srishti-8976 committed:
Delete results directory

Srishti-8976 committed:
Update run-performance.yml

Srishti-8976 committed:
Update run-performance.yml

Srishti-8976 committed:
Update run-performance.yml

Srishti-8976 committed:
Update perfTestRunner.py

Srishti-8976 committed:
Update perfTestRunner.py

Srishti-8976 committed:
Update workflow to commit performance CSV reports

Srishti-8976 committed:
Update run-performance.yml

Srishti-8976 committed:
Update workflow to commit performance CSV reports

Srishti-8976 committed:
Update workflow to commit performance CSV reports

Srishti-8976 committed:
Update

Srishti-8976 committed:
Update workflow to save csv on github

Srishti-8976 committed:
Delete .github/workflows/run-performance.yml

Srishti-8976 committed:
Add workflow to commit performance CSV reports

Srishti-8976 committed:
Add workflow to commit performance CSV reports

Srishti-8976 committed:
Delete .github/workflows/run-performance-csv.yml

Srishti-8976 committed:
Update perf_csv_writer.py

Srishti-8976 committed:
Updating it to save csv on github

Srishti-8976 committed:
Update perf_csv_writer.py

Srishti-8976 committed:
Update perf_csv_writer.py

Srishti-8976 committed:
Updating it

Srishti-8976 committed:
Updating it

Srishti-8976 committed:
Updating it

Srishti-8976 committed:
Create run-performance.yml

Srishti-8976 committed:
Update

Srishti-8976 committed:
Update run-performance.yml

Srishti-8976 committed:
Create .DS_Store

Srishti-8976 committed:
Just update

Srishti-8976 committed:
Revise performance test workflow for FEBRL

Updated workflow to run Zingg FEBRL performance tests and generate CSV reports. Enhanced steps for checking out repositories, installing dependencies, and committing results.

Srishti-8976 committed:
Refactor performance workflow by removing debug step

Removed debug step for verifying input JSON file existence and updated environment variable name for the performance script.

Srishti-8976 committed:
Update input path for performance script

Srishti-8976 committed:
Update input path for FEBRL performance script

Srishti-8976 committed:
Modify performance script input and add debug logs

Updated the performance script to change the input path and added debugging information.

Srishti-8976 committed:
Refactor performance script execution in workflow

Simplified the performance script execution by removing unnecessary commands and directly running the Python script.

Srishti-8976 committed:
Update input path for performance script

Srishti-8976 committed:
Set working directory for performance script

Srishti-8976 committed:
Optimize performance workflow in run-performance.yml

Refactor Python dependency installation and update working directory for performance script execution.

Srishti-8976 committed:
Refactor performance workflow and update steps

Srishti-8976 committed:
Update performance test workflow for FEBRL

Srishti-8976 committed:
Update performance workflow for Zingg FEBRL

Srishti-8976 committed:
Use absolute path for INPUT_FILE

Srishti-8976 committed:
Update performance workflow for repo path changes

Srishti-8976 committed:
Refactor path handling and input validation

Refactor perfTestRunner.py to improve path handling and error checking. Resolve input file as an absolute path and ensure required fields are present in the configuration.

Srishti-8976 committed:
Fix input path for FEBRL performance script

Srishti-8976 committed:
Update input file path for FEBRL performance script

Srishti-8976 committed:
Update input file for FEBRL performance script

Srishti-8976 committed:
Refactor perfTestRunner.py for clarity and structure

Refactor performance test runner to improve clarity and structure. Update error messages and streamline file handling.

Srishti-8976 committed:
Update perf_csv_writer.py

github-actions[bot] committed:
Add FEBRL performance CSV results [auto]

Srishti-8976 committed:
Delete results directory

github-actions[bot] committed:
Add FEBRL performance CSV results [auto]

Srishti-8976 committed:
Update perf_csv_writer.py

Srishti-8976 committed:
Update perfTestRunner.py

Srishti-8976 committed:
Update perfTestRunner.py

Srishti-8976 committed:
Update perf_csv_writer.py

github-actions[bot] committed:
Add FEBRL performance CSV results [auto]

Srishti-8976 committed:
Delete results directory

github-actions[bot] committed:
Add FEBRL performance CSV results [auto]

Srishti-8976 committed:
Create run-performance.yml

github-actions[bot] committed:
Add FEBRL performance CSV results [auto]

Srishti-8976 committed:
Delete .github/workflow directory

github-actions[bot] committed:
Add FEBRL performance CSV results [auto]

Srishti-8976 committed:
Update run-performance.yml

github-actions[bot] committed:
Add FEBRL performance CSV results [auto]

Srishti-8976 committed:
Update perfTestRunner.py

Srishti-8976 committed:
Update perf_csv_writer.py

Srishti-8976 committed:
Update perf_csv_writer.py

Srishti-8976 committed:
Update perfTestRunner.py

github-actions[bot] committed:
Add FEBRL performance CSV results [auto]

Srishti-8976 committed:
Delete results directory

github-actions[bot] committed:
Add FEBRL performance CSV results [auto]

github-actions[bot] committed:
Add FEBRL performance CSV results [auto]

Srishti-8976 committed:
Update run-performance.yml

Srishti-8976 committed:
Update perfTestRunner.py

Srishti-8976 committed:
Update perf_csv_writer.py

github-actions[bot] committed:
Add FEBRL performance CSV results [auto]

Srishti-8976 committed:
Update run-performance.yml

github-actions[bot] committed:
Add FEBRL performance CSV results [auto]

Srishti-8976 committed:
Update run-performance.yml

Srishti-8976 committed:
Update run-performance.yml

Srishti-8976 committed:
Update run-performance.yml

github-actions[bot] committed:
Add FEBRL performance CSV snapshot [auto]

Srishti-8976 committed:
Update run-performance.yml

github-actions[bot] committed:
Add FEBRL performance CSV snapshot [auto]

Srishti-8976 committed:
Delete results directory

github-actions[bot] committed:
Add FEBRL performance CSV snapshot [auto]

github-actions[bot] committed:
Add FEBRL performance CSV snapshot [auto]

padam-prakash commented:
Where are the performance tests actually being executed now?

In this PR, `perfTestRunner.py` has been refactored from a test executor into a JSON-to-CSV converter. The workflow [.github/workflows/run-performance.yml](https://github.com/zinggAI/zingg_performance/blob/patch-2/.github/workflows/run-performance.yml) checks out `zinggAI/zingg_community_performance_reports` and consumes a pre-generated JSON at `INPUT=community_reports/perf_test/perf_test_report/testReport_febrl120K.json`, then writes a timestamped CSV so, it doesn’t run `train/match` here.

If test execution has moved to another repository or workflow, could we document and link to the job that produces the `testReport` JSON? Alternatively, if tests should run in this repo, we might add a job that executes the Zingg scripts to generate the JSON before conversion. Also note that [.github/workflows/runWorkfow_file.yml](https://github.com/zinggAI/zingg_performance/blob/patch-2/.github/workflows/runWorkfow_file.yml) calls `perfTestRunner.py` without setting `INPUT`, so it wouldn’t run successfully.

padam-prakash commented:
remove emojis.

github-actions[bot] committed:
Add FEBRL performance CSV snapshot [auto]

Srishti-8976 committed:
Update perfTestRunner.py

Srishti-8976 committed:
Update perf_csv_writer.py

Srishti-8976 committed:
Update run-performance.yml

Srishti-8976 committed:
Update run-performance.yml

Srishti-8976 committed:
Update run-performance.yml
